diam/sm/sm.go

```diff
@@ -41,10 +41,11 @@ func PrepareSupportedApps(d *dict.Parser) []*SupportedApp {
 // Settings used to configure the state machine with AVPs to be added
 // to CER on clients or CEA on servers.
 type Settings struct {
-	OriginHost  datatype.DiameterIdentity
-	OriginRealm datatype.DiameterIdentity
-	VendorID    datatype.Unsigned32
-	ProductName datatype.UTF8String
+	OriginHost   datatype.DiameterIdentity
+	OriginRealm  datatype.DiameterIdentity
+	HostIPAdress datatype.Address
```

Comment:
Just an aesthetic change that might be important: let's move this field to the bottom of the struct and add comments explaining that it is optional and the use case for it.